frontend/packages/network-attachment-definition-plugin/src/components/network-attachment-definitions/network-attachment-definition.tsx

```diff
@@ -0,0 +1,122 @@
+import * as React from 'react';
+import * as classNames from 'classnames';
+import { sortable } from '@patternfly/react-table';
+import { ListPage, Table, TableData, TableRow } from '@console/internal/components/factory';
+import { Kebab, ResourceKebab, ResourceLink } from '@console/internal/components/utils';
+import { NamespaceModel } from '@console/internal/models';
+import { K8sResourceKindReference } from '@console/internal/module/k8s';
+import { getConfig } from '../../utils';
+import { NetworkAttachmentDefinitionModel } from '../../models';
+import { NetworkAttachmentDefinitionKind } from '../../types';
+
+const { common } = Kebab.factory;
+const menuActions = [...common];
+
+export const NetworkAttachmentDefinitionReference: K8sResourceKindReference =
+  'NetworkAttachmentDefinition';
+
+const tableColumnClasses = [
+  classNames('col-lg-4', 'col-md-4', 'col-sm-6', 'col-xs-6'),
+  classNames('col-lg-4', 'col-md-4', 'hidden-sm', 'hidden-xs'),
+  classNames('col-lg-4', 'col-md-4', 'col-sm-6', 'col-xs-6'),
+  Kebab.columnClass,
+];
+
+const NetworkAttachmentDefinitionsHeader = () => {
+  return [
+    {
+      title: 'Name',
+      sortField: 'metadata.name',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[0] },
+    },
+    {
+      title: 'Namespace',
+      sortField: 'metadata.namespace',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[1] },
+    },
+    {
+      title: 'Type',
+      sortFunc: 'string',
+      transforms: [sortable],
+      props: { className: tableColumnClasses[2] },
+    },
+    {
+      title: '',
+      props: { className: tableColumnClasses[3] },
+    },
+  ];
+};
+NetworkAttachmentDefinitionsHeader.displayName = 'NADTableHeader';
+
+const NetworkAttachmentDefinitionsRow: React.FC<NetworkAttachmentDefinitionsRowProps> = ({
+  obj,
+  index,
+  key,
+  style,
+  namespace,
+}) => {
+  const config = getConfig(obj);
+
+  return (
+    <TableRow id={obj.metadata.uid} index={index} trKey={key} style={style}>
+      <TableData className={classNames(tableColumnClasses[0], 'co-break-word')}>
+        <ResourceLink
+          kind={NetworkAttachmentDefinitionModel.kind}
+          name={obj.metadata.name}
+          namespace={namespace}
+        />
+      </TableData>
+      <TableData className={classNames(tableColumnClasses[1], 'co-break-word')}>
+        <ResourceLink kind={NamespaceModel.kind} name={obj.metadata.namespace} title={namespace} />
+      </TableData>
+      <TableData className={tableColumnClasses[2]}>{config.type || '---'}</TableData>
```

Comment:
I wonder about those `---` for unavailable info. When we have such a case in dashboards, we show `unavailable` string. Shouldn't it be aligned ? @Ranelim @matthewcarleton

Reply:
+1, I think this is a pattern in the console currently but `unavailable` is much better for accessibility.

Reply:
just one more thing, for 4.3 dashboard we are actually using `Not available` instead of `unavailable`. So lets use the newer one. Also it should be styled with `text-secondary` class

Reply:
Done!